vsintegration/src/FSharp.Editor/Hints/InlineTypeHints.fs

```diff
@@ -21,11 +21,19 @@ type InlineTypeHints(parseResults: FSharpParseFileResults, symbol: FSharpMemberO
         // not sure when this can happen
         | None -> []
 
+    let getTooltip _document _cancellationToken =
+        async {
+            // the hardest part
```

Comment:
@majocha I need your help here, you are the expert on this topic :)

So - what do you think is the easiest way to go here? I feel two options:
1) Somehow adopt existing tooltip code we have - all the stuff from WpfFactories and so on. I don't see an easy way of copypasting it here because there is no notion of WPF but I might be missing something
2) Somehow follow what C# does [here](https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/InlineHints/InlineHintHelpers.cs). That also doesn't seem to be very easy, also I wonder how consistent the experience will be in the end if we follow this way.

WDYT?

Reply:
https://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/InlineHints/InlineHint.cs,efca95f740111c98

Looks like the API for inline hints does not allow that much flexibility. `TaggedText` array, so it can display colored text formatted with linebreaks, probably like C# does it. I'd just grab the tooltip from the compiler and display it's text for a start.

Reply:
@majocha So yep, eventually I ended up displaying the most basic info, see the screenshots. But now we can follow up on this, adding colors and links by plugging in respective services.